Formula/avr-gcc@11.rb

```diff
@@ -63,6 +61,14 @@ class AvrGccAT11 < Formula
     end
   end
 
+  # This patch fixes a GCC compilation error on Apple ARM systems by adding
+  # a defintion for host_hooks.  Patch comes from
+  # https://github.com/riscv/riscv-gnu-toolchain/issues/800#issuecomment-808722775
+  patch do
+    url "https://gist.githubusercontent.com/DavidEGrayson/88bceb3f4e62f45725ecbb9248366300/raw/c1f515475aff1e1e3985569d9b715edb0f317648/gcc-11-arm-darwin.patch"
+    sha256 "c4e9df9802772ddecb71aa675bb9403ad34c085d1359cb0e45b308ab6db551c6"
+  end
```

Comment:
So you're sure that without this part, it doesn't work/compile on M1?

Reply:
Yes, I was getting an error about `_host_hooks` not being defined when I tried to build on an M1 machine.

- [Here's the error I saw.](https://github.com/osx-cross/homebrew-avr/issues/237#issuecomment-859816991)
- [Here are other people getting the same error.](https://github.com/riscv/riscv-gnu-toolchain/issues/800#issuecomment-808722775)

Reply:
thanks!